tests/asahirov/test_homework_7.py

```diff
@@ -0,0 +1,191 @@
+import random
+from operator import gt, lt, le, ge
+import time
+
+"""
+Create decorator which would remember answers,
+and if answer was given, do not make a call to a function
+"""
+
+
+def remember_answer(func):
+    answer = {}
+
+    def remember(*args):
+        if args in answer:
+            return answer[args]
+        else:
+            result = func(*args)
+            answer[args] = result
+            return result
```

Comment:
i suggest to search ways to reduce "return" amount and remove nested blocks e,g.

```
if k not in dict:
     dict[k] = res
return dict[k]



```